core/src/test/java/org/apache/accumulo/core/metadata/schema/DataFileValueTest.java

```diff
@@ -0,0 +1,143 @@
+import org.junit.jupiter.api.Test;
+
+public class DataFileValueTest {
+
+  @Test
+  public void testNoTimeOrRanges() {
+    final String json = new DataFileValue(555, 23).encodeAsString();
+    final DataFileValue dfv = DataFileValue.decode(json);
+
+    assertEquals(555, dfv.getSize());
+    assertEquals(23, dfv.getNumEntries());
+    assertEquals(-1, dfv.getTime());
+    assertEquals(0, dfv.getRanges().size());
+  }
+
+  @Test
+  public void testTimeNoRanges() {
+    long time = System.currentTimeMillis();
+    final String json = new DataFileValue(555, 23, time).encodeAsString();
+    final DataFileValue dfv = DataFileValue.decode(json);
+
+    assertEquals(555, dfv.getSize());
+    assertEquals(23, dfv.getNumEntries());
+    assertEquals(time, dfv.getTime());
+    assertEquals(0, dfv.getRanges().size());
+  }
+
+  @Test
+  public void testInfiniteRange() {
+    long time = System.currentTimeMillis();
+    final DataFileValue dfv = new DataFileValue(555, 23, time, List.of(new Range()));
+    assertEquals(1, dfv.getRanges().size());
+    assertEquals(new Range(), dfv.getRanges().get(0));
+
+    // Verify serialization
+    final String json = new DataFileValue(555, 23, time, List.of(new Range())).encodeAsString();
+    final DataFileValue decoded = DataFileValue.decode(json);
+
+    assertEquals(555, decoded.getSize());
+    assertEquals(23, decoded.getNumEntries());
+    assertEquals(time, decoded.getTime());
+    assertEquals(1, decoded.getRanges().size());
+    assertEquals(new Range(), decoded.getRanges().get(0));
+  }
+
+  @Test
+  public void testComplexRange() {
+    final Key key1 = new Key("row1", "cf1", "cq1", 50);
+    final Key key2 = new Key("row2", "cf2", "cq2", 100);
+    final String json = new DataFileValue(555, 23, List.of(new Range(key1, key2))).encodeAsString();
+    final DataFileValue dfv = DataFileValue.decode(json);
+
+    assertEquals(555, dfv.getSize());
+    assertEquals(23, dfv.getNumEntries());
+    assertEquals(-1, dfv.getTime());
+    assertEquals(1, dfv.getRanges().size());
+    assertEquals(new Range(key1, key2), dfv.getRanges().get(0));
+  }
+
+  @Test
+  public void testOverlappingRanges() {
+    long time = System.currentTimeMillis();
+    final DataFileValue dfv =
+        new DataFileValue(555, 23, time, List.of(new Range("abc", "xyz"), new Range("def")));
+    // Second range is part of first range so should collapse into 1
+    assertEquals(1, dfv.getRanges().size());
+    assertEquals(new Range("abc", "xyz"), dfv.getRanges().get(0));
+  }
+
+  @Test
+  public void testMultipleRanges() {
+    long time = System.currentTimeMillis();
+    final DataFileValue dfv =
+        new DataFileValue(555, 23, time, List.of(new Range("abc", "xyz"), new Range("123", "789")));
+
+    // Ranges will be sorted
+    assertEquals(2, dfv.getRanges().size());
+    assertEquals(new Range("123", "789"), dfv.getRanges().get(0));
+    assertEquals(new Range("abc", "xyz"), dfv.getRanges().get(1));
```

Comment:
Metadata data ranges are exclusive,inclusive.  Like [this code](https://github.com/apache/accumulo/blob/30d823def879234b4f390f7b816cef670d477f88/core/src/main/java/org/apache/accumulo/core/dataImpl/KeyExtent.java#L372) converts creates a data range for a tablet. So should test this case, could do the following.


```suggestion
        new DataFileValue(555, 23, time, List.of(new Range("abc", false, "xyz", true), new Range("123", "789")));

    // Ranges will be sorted
    assertEquals(2, dfv.getRanges().size());
    assertEquals(new Range("123", "789"), dfv.getRanges().get(0));
    assertEquals(new Range("abc",false, "xyz",true), dfv.getRanges().get(1));
```